src/js/modules/plant/model.js

```diff
@@ -11,18 +12,50 @@ export default Model.extend({
     height: 0,
     userActivity: false,
   },
+  container: undefined,
 
-  getProjection: function() {
+  constructor({ containerWidth, containerHeight, ...props }, options) {
+    this.container = new BModel({
+      width: containerWidth,
+      height: containerHeight });
+
+    return Model.call(this, props, options);
+  },
+
+  getProjection() {
     return this.manifesto()
       .getProjectionsFor(this.get('objectId'))[
         this.get('currentProjection') || this.get('projection') || 0
       ];
   },
 
-  setProjection: function(at_) {
+  setProjection(at_) {
     const projections = this.get('projections');
     let at = at_ > projections.length ? 0 : at_;
     at = at < 0 ? projections.length : at;
     this.set('projection', at);
   },
+
+  setContainerSize({ width, height }) {
+    this.container.set({ width, height });
+  },
+
+  setPosition({ x, y }) {
+    const { width, height } = this.container.attributes;
```

Comment:
That's not exactly backbone style to use raw `attributes`? Shouldn't do this like `this.container.get('x', 'y')` or something like that?